### ISSUE TITLE
Fail outcome is reported instead of Error when error or exception is raised.

### ISSUE THREAD

**Issue by BernieWhite** [BernieWhite/PSRule#211]
**Description of the issue**

`Fail` is reported as the rule outcome instead of `Error` when a rule or engine error/ exception is raised.

**Expected behaviour**

Errors should be reported as the `Error` outcome.

**Error output**

Capture any error messages and or verbose messages with `-Verbose`.

```text
$r='data'|Invoke-PSRule -Path "$Path\$File" -Outcome all
# Invoke-PSRule : An invalid rule result was returned for Test.Rule.ps1/Test. Conditions must return boolean $True or $False.
#...
#     + FullyQualifiedErrorId : PSRule.Runtime.InvalidRuleResult,Invoke-PSRule

$r|select *
# Outcome        : Fail
# OutcomeReason  : Processed
#...
# RuleId         : Test.Rule.ps1/Test
# RuleName       : Test
```

**Module in use and version:**

- Module: PSRule
- Version: **0.7.0-B190633**

**Additional context**

#187
